DEVELOPING_A_RUNTIME.md

```diff
@@ -16,146 +36,161 @@ A **required** exported constant that decides which version of the Runtime API t
 
-```ts
-import { PrepareCacheOptions } from '@vercel/build-utils'
+export async function prepareCache(options: PrepareCacheOptions) {
+  // Create a mapping of file names and `File` object instances to cache here…
 
-export prepareCache(options: PrepareCacheOptions) {
-  return { 'path-to-file': File }
+  return {
+    'path-to-file': File,
+  };
 }
 ```
 
-### `shouldServe`
+### `shouldServe()`
 
-An **optional** exported function that is only used by `vercel dev` in [Vercel CLI](https:///download) and indicates whether a [Runtime](https://vercel.com/docs/runtimes) wants to be responsible for building a certain request path.
+An **optional** exported function that is only used by `vercel dev` in [Vercel
+CLI](https://vercel.com/download) and indicates whether a
+[Runtime](https://vercel.com/docs/runtimes) wants to be responsible for responding
+to a certain request path.
 
-```js
-shouldServe({
-  entrypoint: String,
-  files: Files,
-  config: Object,
-  requestPath: String,
-  workPath: String
-}) : Boolean
-```
+**Example:**
 
-If you are using TypeScript, you can import the types for each of these functions by using the following:
+```typescript
+import { ShouldServeOptions } from '@vercel/build-utils';
 
-```ts
-import { ShouldServeOptions } from '@vercel/build-utils'
+export async function shouldServe(options: ShouldServeOptions) {
+  // Determine whether or not the Runtime should respond to the request path here…
 
-export shouldServe(options: ShouldServeOptions) {
-  return Boolean
+  return options.requestPath === options.entrypoint;
 }
 ```
 
-If this method is not defined, Vercel CLI will default to [this function](https://github.com/vercel/vercel/blob/52994bfe26c5f4f179bdb49783ee57ce19334631/packages/now-build-utils/src/should-serve.ts).
+If this function is not defined, Vercel CLI will use the [default implementation](https://github.com/vercel/vercel/blob/52994bfe26c5f4f179bdb49783ee57ce19334631/packages/now-build-utils/src/should-serve.ts).
 
-### Runtime Options
+### `startDevServer()`
 
-The exported functions [`analyze`](#analyze), [`build`](#build), and [`prepareCache`](#preparecache) receive one argument with the following properties.
+An **optional** exported function that is only used by `vercel dev` in [Vercel
+CLI](https://vercel.com/download). If this function is defined, Vercel CLI will
+**not** invoke the `build()` function, and instead invoke this function for every
+HTTP request, so it is an opportunity to provide a faster development experience
+rather than going through the entire `build()` process that is used in production.
 
-**Properties:**
+This function is invoked _once per HTTP request_ and is expected to spawn a child
+process which creates an HTTP server that will execute the entrypoint code when
+an HTTP request is received. This child process is _single-serve_ (only used for
+a single HTTP request), so after `vercel dev` completes the HTTP request to the
```

Comment:
"completes the HTTP request to the child process" sounds odd? Can this be phrased differently? 

Something like "After the HTTP response, vercel dev sends a shut down signal to the child process"